### PR TITLE
docs: document stickiness

### PR DESCRIPTION
**Pull request by thomasheartman**
## overview

This PR adds a doc page on stickiness, documenting what it is and how it's calculated. It also has a section on custom stickiness.

## to discuss

I definitely need some input on this document. I've cobbled it together from playing with the demo app and sourcing information from other parts of the documentation, so it might not all be correct.

I have also removed some old pages that seemed to hang around for no apparent reason. I realize now that this _may_ have been to keep the links alive. If that's the case, can we instead redirect requests to those urls or should I revive the files?

## render

For reference, the page looks a little like this when rendered (though firefox messed up part of the screenshot, it seems 🙈):

image

### CODE REVIEW

**Comment by chriswk** [comment]
Speaking of render. I'll be looking into deploying doc PRs to a vercel PR deployment so we can all see what's happening, but for now, the included render is much appreciated :)

File: website/docs/advanced/stickiness.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+id: stickiness
+title: Stickiness
+---
+
+_Stickiness_ is how Unleash guarantees that the same user gets the same features every time. Stickiness is useful in any scenario where you want to either show a feature to only a subset of users or give users a variant of a feature.
+
+## Calculation
+
+By default, Unleash calculates stickiness based on the user id and the group id. If the user id is unavailable, it falls back to using the session id instead. It hashes these values to a number between 0 and 100. This number is what determines whether a user will see a specific feature or variant. Because the process is deterministic, the same user will always get the same number.
```
Comment (chriswk) [inline website/docs/advanced/stickiness.md:10]: Do we want to point out that we use the murmur hash, or is that not interesting here?

File: website/docs/advanced/stickiness.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+## Consistency
+
+Because the number assigned to a user won't change, Unleash also guarantees that the a user will keep seeing the same features even if certain other parameters change.
+
+For instance: When using the [gradual rollout activation strategy](../user_guide/activation-strategies.md#gradual-rollout), any user whose number is less than or equal to the rollout percentage will see the feature. This means that the same users will keep seeing the feature even as you increase the percentage of your user base that sees the feature.
+
+## Custom stickiness (beta)
+
+:::info
+Custom stickiness is available Unleash Enterprise v4 and later.
```
Comment (chriswk) [inline website/docs/advanced/stickiness.md:23]:
```suggestion
Custom stickiness is available starting from Unleash Enterprise v4.
```

File: website/docs/api/client/feature-toggles-api.md [hunk 3]
```diff
@@ -117,7 +117,7 @@ Used to fetch details about a specific feature toggle. This is mainly provided t
 
 > This is a unleash-enterprise feature
 
-Strategy definitions may also contain a `constraints` property. Strategy constraints is a feature in Unleash which work on context fields, which is defined as part of the [Unleash Context](/unleash_context). The purpose is to define a set of rules where all needs to be satisfied in order for the activation strategy to . A [high level description](https://www.unleash-hosted.com/articles/strategy-constraints) of it is available online.
+Strategy definitions may also contain a `constraints` property. Strategy constraints is a feature in Unleash which work on context fields, which is defined as part of the [Unleash Context](../../user_guide/unleash-context.md). The purpose is to define a set of rules where all needs to be satisfied in order for the activation strategy to . A [high level description](https://www.unleash-hosted.com/articles/strategy-constraints) of it is available online.
```
Comment (chriswk) [inline website/docs/api/client/feature-toggles-api.md:120]:
```suggestion
Strategy definitions may also contain a `constraints` property. Strategy constraints is a feature in Unleash which work on context fields, which is defined as part of the [Unleash Context](../../user_guide/unleash-context.md). The purpose is to define a set of rules where all needs to be satisfied in order for the activation strategy to evaluate to true. A [high level description](https://www.unleash-hosted.com/articles/strategy-constraints) of it is available online.
```

**Review by chriswk — APPROVED** [review]
I like this as a first version.

I'm good with starting here, and then letting it evolve as we find the gaps.

File: website/docs/advanced/stickiness.md (hunk 1 above)
Comment (thomasheartman) [reply website/docs/advanced/stickiness.md:10]: I don't know, actually, but I like the suggestion. I'll add it in! 🙋🏾

**Comment by thomasheartman** [comment]
@chriswk Suh-weet, you added previews 🤩  Super excited for that 👏🏾 

Anyway, I accepted your suggestions and added a little note about the MurmurHash hash function. I'll roll it out once the checks pass 🚂

**Pull request merged**